extension/src/service/web-component-service.js

```diff
@@ -1,6 +1,6 @@
 const fetch = require('node-fetch')
```

Comment:
Benefit is that you have it organized :) We can put them all in one folder, but then it's lots of files at once. That's the idea of packages in java as well. There are several ways to group the files into packages, and in our case, I see grouping them by type as most suitable.

Reply:
Here I will try to put my ideas so it's just matter of taste and also it's not a big issue to separate them by types, but also you need to consider that the extension module is a microservice, it could make sense to separate them if you have lots of files in a folder but it's not a case here, IMO it's layering the implementation unnecessarily **outside** of the payment handler package is wrong, which is based on the file namings, the file name already has this additional layer information so it could be in the same package or could be inside a sub-package called service but **not outside**. I guess for the web-components-service it was also in the wrong package before 🤦 Anyways we could go back that package organization inside the extension module later separately.

IMO in java packages idea is not that way like grouping by types, for example, in java SDK packages grouping is not based on the internal layers, it's rather separated by development functionality, for example, take `java. util.concurrent` package into consideration, the package groups the concurrency utilities that you could use, so in that package, most of the things are in one place if it does have lots of different use cases or code-sharing only then java packages are separated inside that package for example:  `java.util.concurrent.atomic`, so the java package groupings are not putting additional folders like `java.util.concurrent.future` because Future is used all in the same level of the `java.util.concurrent` package.

Reply:
You're right that the grouping by type it's not very good, but I don't see a better way here. So do you suggest to put everything under `src` folder? Isn't it too many files in one folder if we mix `paymentHandlers`, `validators`, `service` etc.?

Reply:
I will skip this comment if you could not see the better way, also according to my other comments there is no klarna service, so the package will only include web-components-service.